### PR TITLE
Fix v5 changes and exceptions, enforce min/max values

### PR DESCRIPTION
**Pull request by mitchcapper**
I noticed some startup exceptions that might have caused some instability you were seeing.  As loupedeck swallows exceptions by default without a debugger attached they would not be seen.

Minor note: I did not check your streamdeck code to see if it is an artifact of porting that but:
SingleBaseAdjustment.cs was normalizing the min/max values to zero then subtracting it out when displayed.   I would recommend storing the true value with an adjustment offset that you add/sub only when directly sending to loupedeck (if needed).  I made this change, but then noticed it doesn't actually seem to be used anywhere.  

Also added min/max values enforcement so you can't scroll beyond the VM limits.

There might be a lingering issue if the values and amount bar are not supposed to be rotated (if so let me know and I can see about fixing).

### CODE REVIEW

**Comment by XeroxDev** [comment]
I've changed this PR to merge into beta instead of main.

File: VoiceMeeterPlugin/Commands/Bases/BooleanBaseCommand.cs [hunk 1]
```diff
@@ -124,12 +120,13 @@ private void GetNewSettings()
                             $"{(this.IsStrip ? "Strip" : "Bus")}[{hiIndex + this.Offset}].{this.Command}") == 1;
                 }
             }
-
-            this.ActionImageChanged();
+            if (loaded)
```
Comment (XeroxDev) [inline VoiceMeeterPlugin/Commands/Bases/BooleanBaseCommand.cs:123]: Could you please add `this.` (due to the editorconfig rules)

File: VoiceMeeterPlugin/Commands/Bases/BooleanBaseCommand.cs [hunk 2]
```diff
@@ -124,12 +120,13 @@ private void GetNewSettings()
...
+                this.ActionImageChanged();
         }
-
+        protected bool loaded;
```
Comment (XeroxDev) [inline VoiceMeeterPlugin/Commands/Bases/BooleanBaseCommand.cs:126]: It would be good, if you could move this variable to the others, also respect the naming convention please.

File: VoiceMeeterPlugin/Commands/Bases/BooleanBaseCommand.cs [hunk 3]
```diff
@@ -124,12 +120,13 @@ private void GetNewSettings()
...
         protected override Boolean OnLoad()
-        {
+        {
+            loaded = true;
```
Comment (XeroxDev) [inline VoiceMeeterPlugin/Commands/Bases/BooleanBaseCommand.cs:129]: Could you please add `this.` (due to the editorconfig rules)

File: VoiceMeeterPlugin/Commands/Bases/SingleBaseAdjustment.cs [hunk 4]
```diff
@@ -86,12 +82,14 @@ private void GetNewSettings()
                     (Int32)Remote.GetParameter(
                         $"{(this.IsStrip ? "Strip" : "Bus")}[{hiIndex + this.Offset}].{this.Command}");
             }
-
+
+            if (loaded)
```
Comment (XeroxDev) [inline VoiceMeeterPlugin/Commands/Bases/SingleBaseAdjustment.cs:86]: Could you please add `this.` (due to the editorconfig rules)

File: VoiceMeeterPlugin/Commands/Bases/SingleBaseAdjustment.cs [hunk 5]
```diff
@@ -86,12 +82,14 @@ private void GetNewSettings()
...
+
+            if (loaded)
             this.AdjustmentValueChanged();
         }
-
+        protected bool loaded;
         protected override Boolean OnLoad()
-        {
+        {
+            loaded = true;
```
Comment (XeroxDev) [inline VoiceMeeterPlugin/Commands/Bases/SingleBaseAdjustment.cs:92]: Could you please add `this.` (due to the editorconfig rules)

File: VoiceMeeterPlugin/Commands/Bases/SingleBaseAdjustment.cs [hunk 6]
```diff
@@ -86,12 +82,14 @@ private void GetNewSettings()
                     (Int32)Remote.GetParameter(
                         $"{(this.IsStrip ? "Strip" : "Bus")}[{hiIndex + this.Offset}].{this.Command}");
             }
-
+
+            if (loaded)
             this.AdjustmentValueChanged();
         }
-
+        protected bool loaded;
```
Comment (XeroxDev) [inline VoiceMeeterPlugin/Commands/Bases/SingleBaseAdjustment.cs:89]: It would be good, if you could move this variable to the others, also respect the naming convention please.

**Review by XeroxDev — CHANGES_REQUESTED** [review]
I've added a few comments with change requests.

**Comment by mitchcapper** [comment]
re: is ValueToAdd not actually needed

To review the ValueToAdd effect we will look at 2 situations:

- The old code which applies a correction factor so the range is always starting no lower than 0
  
- New code with it removed completely
  

**tldr***: The result of review was removing had no effect due to the min/max value components are always de-corrected before being used.

### Details

Given the gain control with the old code:
`Actual: minValue -65 max value = 12`

These into the constructor change to:
 `minValue = 0, maxValue = 77, ValueToAdd = 65`

(ValueToAdd is added to both min and max inputs)

`minValue` and `MaxValue` and `ValueToAdd` were only used in one place:

`protected override BitmapImage GetAdjustmentImage(String actionParameter, PluginImageSize imageSize)`

    it calculates the percentage for the bar using

`var percentage = (currentValue + this.ValueToAdd - this.MinValue) / (this.MaxValue - this.MinValue) * 100;`

Lets assume it is at 25% the max value the real level is `-45.75`

With the original code:

`(-45.75 + 65 - 0) / (77 - 0) * 100 = 19.25 / 0.77 = 25%`

New code:

`-45.75 - -65 / (12- -65)*100 = 19.25 / 0.77 = 25%`

The new code formula is simplified to:

`var percentage = (currentValue - this.MinValue) / (this.MaxValue - this.MinValue) * 100;`

The reason the percentages are the same is more apparent with the original formula rewritten where we replace Max/Min values with their orig value + offsets (what they are and reduce this):

`(currentValue + this.ValueToAdd - [origMinValue+this.ValueToAdd]) / ( [origMaxValue+this.ValueToAdd] - [origMinValue + this.ValueToAdd]) * 100;`

`(currentValue + this.ValueToAdd - origMinValue - this.ValueToAdd) / (origMaxValue + this.ValueToAdd - origMinValue - this.ValueToAdd) * 100`

`(currentValue - origMinValue) / (origMaxValue - origMinValue) * 100`

As the ValueToAdd completely drops out, and is not used anywhere else, it is not needed.

The min/max/value to add are not used in any way when the user sets the value from the device or when a new value comes in from VM.

**Review by XeroxDev — APPROVED** [review]
Nice work! Now it's mergeable.

**Comment by XeroxDev** [comment]
This is getting closed due to separation (see #7)
